clients/spark/core/src/main/scala/io/treeverse/clients/GCBackupAndRestore.scala

```diff
@@ -0,0 +1,180 @@
+package io.treeverse.clients
+
+import org.apache.hadoop.conf.Configuration
+
+import scala.util.control.Breaks._
+import org.apache.hadoop.tools.DistCp
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+import java.net.URI
+
+class GCBackupAndRestore {}
+
+object GCBackupAndRestore {
+  val S3AccessKeyName = "fs.s3a.access.key"
+  val S3SecretKeyName = "fs.s3a.secret.key"
+  val AzureStorageAccountKeyName = "fs.azure.account.key"
+  lazy val spark = SparkSession.builder().appName("GCBackupAndRestore").getOrCreate()
+
+  /** Validates that args are the following:
+   *  1. address of parquet that includes the relative paths of files to backup or restore, created by a gc run
+   *     2. src: the namespace to backup or restore objects from/to (i.e. repo storage namespace, or an external location compatibly)
+   *     3. backup/restore destination: the namespace to backup or restore objects from/to (i.e. an external location or repo storage namespace compatibly)
+   *     4. Object storage type: "s3" or "azure"
+   */
+  def validateArgs(args: Array[String]): Unit = {
+    if (args.length != 4) {
+      Console.err.println(
+        "Usage: ... <objects list location> <src namespace> <dst namespace> <storage type>"
+      )
+      System.exit(1)
+    }
+
+    val objectsListPath = args(0)
+    if (!objectsListPath.endsWith("parquet")) {
+      Console.err.println(
+        "<objects list path> must be of type parquet"
+      )
+      System.exit(1)
+    }
+
+    val storageType = args(3)
+    if (
+      !storageType
+        .equals(StorageUtils.StorageTypeS3) && !storageType.equals(StorageUtils.StorageTypeAzure)
+    ) {
+      Console.err.println(
+        "Invalid <storage type>, must be either \"s3\" or \"azure\""
+      )
+      System.exit(1)
+    }
+  }
```

Comment:
Not sure I like it.  I deploy a client, a year later Treeverse add GC support for GCS, and now my perfectly good client refuses to run.

If we must have this, I'd be happier to include gcs in there as well... even though it will fail later.

Reply:
I will remove it and let it fail in another storage type validation. I don't like the option of adding GCS at this point